### PR TITLE
Testing events

### PR DESCRIPTION
**Pull request by NickTUD**
Wrote some simple tests for the events, but kinda ran out of time.

### CODE REVIEW

File: core/test/cg/group4/game_logic/stroll/events/TestStrollEventTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        TestStrollEvent spy = Mockito.spy(cTestEvent);
+
+        spy.taskCompleted();
+        Mockito.verify(spy).doTask();
+    }
+
+    @Test
+    public void taskCompletedComplete() {
+        cTestEvent.cTasksCompleted = 20;
+        TestStrollEvent spy = Mockito.spy(cTestEvent);
```
Comment (sirpolomotro) [inline core/test/cg/group4/game_logic/stroll/events/TestStrollEventTest.java]: Cant the `cTestEvent` already be a spy by adding this to the constructor. You already wrote this code twice.

File: core/src/cg/group4/game_logic/stroll/events/TestStrollEvent.java [hunk 2]
```diff
@@ -189,7 +195,7 @@ public final int getReward() {
     }
 
     @Override
-    public final void update(final Observable o, final Object arg) {
+    public void update(final Observable o, final Object arg) {
```
Comment (sirpolomotro) [inline core/src/cg/group4/game_logic/stroll/events/TestStrollEvent.java]: Why is the final removed here? So far it did not need to be removed.

**Comment by sirpolomotro** [comment]
Pretty basics tests, but good work.

**Comment by Surskit** [comment]
Other than the things sirpolomotro already said, I see nothing wrong with it. +1

**Comment by NickTUD** [comment]
Implemented some more tests, but got stuck on the clearEvent() method for example.

**Pull request merged**